ynr/apps/parties/importer.py

```diff
@@ -0,0 +1,324 @@
+                deregistered_date.group(1), dayfirst=True
+            )
+        else:
+            raise ValueError(
+                "Unknown deregistrtion date for '{}'".format(
+                    self["RegulatedEntityName"]
+                )
+            )
+
+    @property
+    def register(self):
+        if not self["RegisterName"] and "PPm" in self.ec_id:
+            # Minor parties don't always need to be on a register
+            return None
+        expected_names = ["Northern Ireland", "Great Britain"]
+        if self["RegisterName"] not in expected_names:
+            raise ValueError(
+                "{} is an unknown register for party {}".format(
+                    self["RegisterName"], self.ec_id
+                )
+            )
+        if self["RegisterName"] == "Northern Ireland":
+            return "NI"
+        else:
+            return "GB"
+
+    @property
+    def registration_status(self):
+        if self["RegistrationStatusName"] not in ["Deregistered", "Registered"]:
+            raise ValueError(
+                "{} is an unknown registration status".format(
+                    self["RegistrationStatusName"]
+                )
+            )
+        return self["RegistrationStatusName"]
+
+    def parse_date(self, date_str):
+        timestamp = re.match(r"\/Date\((\d+)\)\/", date_str).group(1)
+        dt = datetime.fromtimestamp(int(timestamp) / 1000.)
+        return dt.strftime("%Y-%m-%d")
+
+
+class ECEmblem:
+    def __init__(self, party, emblem_dict):
+        self.party = party
+        self.emblem_dict = emblem_dict
+
+    def download_emblem(self):
+        url = "{}/{}".format(EC_EMBLEM_BASE, self.emblem_dict["Id"])
+        ntf = NamedTemporaryFile(delete=False)
+        r = requests.get(url)
+        with open(ntf.name, "wb") as f:
+            f.write(r.content)
+        return ntf.name
+
+    def save(self):
+        existing_emblem = PartyEmblem.objects.filter(
+            ec_emblem_id=self.emblem_dict["Id"], party=self.party
+        )
+        if existing_emblem.exists():
+            return (existing_emblem.first(), False)
+
+        image_file_name = self.download_emblem()
+        mime_type = magic.Magic(mime=True).from_file(image_file_name)
+        extension = mimetypes.guess_extension(mime_type.decode("utf8"))
+        if extension == ".html":
+            #  Something has gone wrong here, maybe the EC site is down
+            # or the file is raising a 404 for some reason.
+            # Either way, we don't want this file to be saved
+            return
+        filename = "Emblem_{}{}".format(self.emblem_dict["Id"], extension)
+
+        emblem = PartyEmblem.objects.create(
+            ec_emblem_id=self.emblem_dict["Id"],
+            party=self.party,
+            description=self.emblem_dict["MonochromeDescription"],
+            default=self.get_default(),
+        )
+
+        emblem.image.save(filename, open(image_file_name, "rb"))
```

Comment:
Should we clean up the temp file `image_file_name` at the end here? I reckon given we don't cycle the YNR servers we want to avoid accumulating temp files.